source/MRTest/MRPointCloudVariadicOffsetTests.cpp

```diff
@@ -216,4 +216,42 @@ TEST( MRMesh, WeightedClosed )
     ASSERT_FLOAT_EQ( maxDiff, 1.f );
 }
 
+TEST( MRMesh, findClosestWeightedMeshPointSharpAngle )
+{
+    Triangulation t{
+        { 0_v, 2_v, 1_v },
+        { 0_v, 1_v, 3_v }
+    };
+    VertCoords vs{
+        { 1, 0, 0 },        //0_v
+        { 0, 0, 0 },        //1_v
+        { 0.5, 0.5, 1 },    //2_v
+        { -0.5, -0.5, 1 }   //3_v
+    };
+    auto mesh = Mesh::fromTriangles( std::move( vs ), t );
+
+    DistanceFromWeightedPointsComputeParams params;
+    params.pointWeight = [&]( VertId ) { return 0.05f; };
```

Comment:
Does it work with different weights at each point?

Reply:
Added test for different weights